pkg/issuer/acme/setup.go

```diff
@@ -283,20 +283,24 @@ func (a *Acme) Setup(ctx context.Context) error {
 	account, registeredEmail, err := ensureEmailUpToDate(ctx, cl, account, specEmail)
 	if err != nil {
 		reason = errorAccountUpdateFailed
-		msg = messageAccountUpdateFailed + err.Error()
 		log.Error(err, "failed to update ACME account")
-		a.recorder.Event(a.issuer, corev1.EventTypeWarning, errorAccountUpdateFailed, msg)
 
 		acmeErr, ok := err.(*acmeapi.Error)
 		// If this is not an ACME error, we will simply return it and retry later
 		if !ok {
+			msg = messageAccountUpdateFailed + err.Error()
+			a.recorder.Event(a.issuer, corev1.EventTypeWarning, errorAccountUpdateFailed, msg)
 			return err
 		}
 
+		msg = messageAccountUpdateFailed + fmt.Sprint(acmeErr.StatusCode)
```

Comment:
Why do we want to include only the `StatusCode` in this message, instead of the `err.Error()` message?

Reply:
If we use err.Error(), the error message looks like this:

`Failed to update ACME account:500 : `

I felt it is not formatted well with colon and space at the end of the error string.

With `acmeErr.StatusCode`, the error message looks similar, but better formatted `Failed to update ACME account:500`.

Please provide your suggestions!

Reply:
This is the code that creates this error: https://cs.opensource.google/go/x/crypto/+/eccd6366:acme/types.go;l=111-120
Are we certain that it will always only contain a `StatusCode`?
It might also contain a `ProblemType` and `Detail` info right?

Reply:
Ah.. I missed that. In the test cases, only Status was set in the ACME error and other fields are not. So, the output looked unformatted. I think I need to revert the changes in setup.go and correct the test cases!

Reply:
@inteon , Done!